tests/test_protocol_optimization.py

```diff
@@ -0,0 +1,119 @@
+from optimize.protocol import (
+    rebalance_initial_protocol,
+    log_weights_from_mixture,
+    linear_u_kn_interpolant,
+    construct_work_stddev_estimator,
+)
+
+from pymbar import MBAR
+from pymbar.testsystems import HarmonicOscillatorsTestCase
+
+import numpy as np
+from scipy.special import logsumexp
+
+np.random.seed(2021)
+
+
+def test_rebalance_initial_protocol():
+    """Integration test: assert that protocol optimization improves run-to-run variance in free energy estimates"""
+    initial_protocol = np.linspace(0, 1, 64)
+    mbar = simulate_protocol(initial_protocol, seed=2021)
+    new_protocol = rebalance_initial_protocol(
+        initial_protocol,
+        mbar.f_k,
+        mbar.u_kn,
+        mbar.N_k,
+        work_stddev_threshold=0.25,
+    )
+
+    # for a fair comparison, re-run initial protocol with same number of lambda windows as new_protocol
+    new_K = len(new_protocol)
+    old_protocol = np.linspace(0, 1, new_K)
+
+    # run several replicates to measure run-to-run variability
+    n_replicates = 10
+    old_mbars = []
+    new_mbars = []
+    for i in range(n_replicates):
+        old_mbars.append(simulate_protocol(old_protocol, seed=i))
+        new_mbars.append(simulate_protocol(new_protocol, seed=i))
+
+    old_stddev = np.std([mbar.f_k[-1] for mbar in old_mbars])
+    new_stddev = np.std([mbar.f_k[-1] for mbar in new_mbars])
+
+    print(f"empirical free energy estimate stddev across {n_replicates} replicates, with # of lambda windows = {new_K}")
+    print(f"\told stddev: {old_stddev:.3f}")
+    print(f"\tnew stddev: {new_stddev:.3f}")
```

Comment:
Could test protocol using \sum_k BAR(protocol[k], protocol[k+1]), rather than MBAR(protocol).f_k[-1], to more closely match deployment conditions